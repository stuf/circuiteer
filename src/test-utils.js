import { render as renderRtl } from '@testing-library/react';
import { format } from 'util';
import { Provider } from 'react-redux';
import { preloadedState } from './core/preloaded';
import * as reducers from './state';
import { configureStore } from '@reduxjs/toolkit';

import './test/index';

//

export function render(vdom, { initialState, store, ...renderOptions } = {}) {
  const rootStore =
    store ||
    configureStore({
      preloadedState: initialState || preloadedState,
      reducer: reducers,
    });

  function Wrapper({ children }) {
    return <Provider store={rootStore}>{children}</Provider>;
  }

  return renderRtl(vdom, { wrapper: Wrapper, ...renderOptions });
}

//

export function suppressPropTypeWarnings() {
  const err = console.error.bind(console.error);

  beforeAll(() => {
    console.error = (msg, ...rest) => {
      const errorMsg = format(msg, ...rest);
      if (!errorMsg.toString().includes('Warning: Failed prop type')) {
        err(errorMsg);
      }
    };
  });

  afterAll(() => {
    console.error = err;
  });
}
