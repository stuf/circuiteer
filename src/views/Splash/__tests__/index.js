import { render } from 'test-utils';
import { Splash } from '../index';

test('Splash', () => {
  const { container } = render(<Splash />);

  // expect(container).toMatchSnapshot();
});
