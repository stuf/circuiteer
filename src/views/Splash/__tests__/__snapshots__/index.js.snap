// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash 1`] = `
<div>
  <div
    class="absolute left-0 bottom-0 px-4 py-2 text-xs text-gray-500 flex space-x-1"
  >
    <span>
      @etm12/circuiteer
    </span>
    <span>
      master
    </span>
    <span>
      f9688485
    </span>
  </div>
  <div
    class="h-full w-full relative text-center"
  >
    <div
      class="absolute left-1/2 top-1/2 transform -translate-x-1/2 -translate-y-1/2"
    >
      <div
        class="flex flex-col space-y-4"
      >
        <svg
          class="w-64"
          viewBox="0 0 500 500"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs>
            <path
              d="M199 0l198.3 114.5v229L199 458 .7 343.5v-229z"
              id="a"
            />
          </defs>
          <g
            fill="none"
            fill-rule="evenodd"
          >
            <g
              transform="translate(52 22)"
            >
              <use
                fill="#D8126F"
                href="#a"
                stroke="#D8126F"
                stroke-width="16"
              />
              <use
                href="#a"
                stroke="#FFF"
                stroke-width="6"
              />
            </g>
            <path
              d="M251 79.5l149 86v172l-149 86-149-86v-172l149-86z"
              fill="#D8126F"
              stroke="#FFF"
              stroke-width="6"
            />
            <path
              d="M169.5 182c24.4 0 48.3.4 72.2 1.4.2-12.4 1-22 2-28.9 2.4-.3 5.7-.5 9.4-.5 3 0 6.2.5 9.2 1.5 0 9-.5 18.7-1.2 28.7 24.9 1 49.5 2.7 73.9 4.7v7.2c0 4-.5 6.8-1 8.3-69.2 0-124.2-2-165-6-.8-1.3-1-4-1-8.2 0-2.8.5-5.5 1.5-8.3zm46.3 25.1c4.5 1.5 9 4.8 13.7 10-7 10.5-15.5 21.2-25.4 32 8 7.6 16.2 15.6 24.9 23.6 17.4-15 34.3-29 51-42.1 4.5 3.2 8.5 7 12.2 11-25.9 25.9-55 50.5-87.4 74.5a35 35 0 01-11.4-12c7.5-7 15.2-13.7 22.6-20.2a124.9 124.9 0 01-35.8-33.4c10.4-15 22.4-29.4 35.6-43.4zm94.3 55.6a36 36 0 0111 12.2c-29.9 26-60.8 50.7-92.9 74.1-5-3.7-8.7-8-11.4-12.7 29.3-27 60.5-51.6 93.3-73.6zm-8 46.7c10.3 8.4 19.7 17 28.2 25.6-3 4.3-7.5 7.8-13.5 10.3-11-8-20.4-16.7-28.8-25.7 4.4-4 9.2-7.3 14.1-10.2z"
              fill="#FFF"
            />
          </g>
        </svg>
        <div>
          @etm12/circuiteer
        </div>
      </div>
    </div>
  </div>
</div>
`;
